### PR TITLE
CIRC-1290 Handling title level request notices (scheduled notices)

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose: 
- Handle title level request notices (scheduled notices);
- Create request expiration scheduled notices for TLR;
- Send TLR scheduled notices;

## Approach
- implement TLR scheduled notices handling;
- update logic for scheduled notices sending;
- cover the changes by tests;

Resolves: [CIRC-1290](https://issues.folio.org/browse/CIRC-1290)

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/CreateRequestService.java [hunk 1]
```diff
@@ -68,11 +72,14 @@ public CreateRequestService(CreateRequestRepositories repositories,
...
       .thenCompose(r -> r.after(manualBlocksValidator::validate))
       .thenApply(r -> errorHandler.handleValidationResult(r, manualBlocksValidator.getErrorType(), result))
+      .thenComposeAsync(r -> r.after(when(this::shouldCheckInstance, this::checkInstance, this::doNothing)))
       .thenComposeAsync(r -> r.after(when(this::shouldCheckItem, this::checkItem, this::doNothing)))
       .thenComposeAsync(r -> r.after(when(this::shouldCheckPolicy, this::checkPolicy, this::doNothing)))
       .thenComposeAsync(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration,
         RequestAndRelatedRecords::withTimeZone))
       .thenApply(r -> r.next(errorHandler::failWithValidationErrors))
+      .thenComposeAsync(r -> r.combineAfter(configurationRepository::lookupTlrSettings,
+        RequestAndRelatedRecords::withTlrSettingsConfiguration))
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/CreateRequestService.java:82]: let's save tlrSettings to request entity

File: src/main/java/org/folio/circulation/domain/RequestAndRelatedRecords.java [hunk 2]
```diff
@@ -11,26 +12,28 @@
...
+  private final TlrSettingsConfiguration tlrSettingsConfiguration;
 
   private RequestAndRelatedRecords(
     Request request,
     RequestQueue requestQueue,
     RequestPolicy requestPolicy,
     MoveRequestRecord moveRequestRecord,
-    ZoneId timeZone) {
+    ZoneId timeZone,
+    TlrSettingsConfiguration tlrSettingsConfiguration) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/RequestAndRelatedRecords.java:24]: remove

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 3]
```diff
@@ -70,9 +70,12 @@ public static JsonObject createLoanNoticeContext(Loan loan) {
   public static JsonObject createRequestNoticeContext(Request request) {
     JsonObject requestNoticeContext = new JsonObject()
       .put(USER, createUserContext(request.getRequester()))
-      .put(ITEM, createItemContext(request.getItem()))
       .put(REQUEST, createRequestContext(request));
 
+    if (request.getItem() != null) {
+      requestNoticeContext.put(ITEM, createItemContext(request.getItem()));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:76]: please leave a comment why item may absent

add check item.isFound()

File: src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java [hunk 4]
```diff
@@ -534,6 +530,195 @@ void scheduledNoticesShouldNotBeSentOrDeletedWhenPatronNoticeRequestFails() {
...
+    scheduledNoticeProcessingClient.runRequestNoticesProcessing();
+
+    verifyNumberOfSentNotices(1);
+    verifyNumberOfScheduledNotices(0);
+    verifyNumberOfPublishedEvents(NOTICE, 1);
+    verifyNumberOfPublishedEvents(NOTICE_ERROR, 0);
+  }
+
+  @Test
+  void titleLevelRequestExpirationNoticeShouldNotBeCreatedWithDisabledTlr() {
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java:558]: Add test for missing settings

File: src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java [hunk 5]
```diff
@@ -534,6 +530,195 @@ void scheduledNoticesShouldNotBeSentOrDeletedWhenPatronNoticeRequestFails() {
...
+    final var requestExpiration = LocalDate.of(localDate.getYear(),
+      localDate.getMonthValue(), localDate.getDayOfMonth());
+    requestsFixture.place(buildTitleLevelRequest(requestExpiration));
+
+    verifyNumberOfScheduledNotices(0);
+  }
+
+  @ParameterizedTest
+  @EnumSource(TlrFeatureStatus.class)
+  void itemLevelRequestExpirationNoticeShouldBeCreatedAndSentRegardlessTlr(
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java:587]:
```suggestion
  void itemLevelRequestExpirationNoticeShouldBeCreatedAndSentRegardlessTlrSettings(
```

File: src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java [hunk 6]
```diff
@@ -534,6 +530,195 @@ void scheduledNoticesShouldNotBeSentOrDeletedWhenPatronNoticeRequestFails() {
...
+    scheduledNoticeProcessingClient.runRequestNoticesProcessing();
+
+    verifyNumberOfSentNotices(2);
+    verifyNumberOfScheduledNotices(0);
+    verifyNumberOfPublishedEvents(NOTICE, 2);
+    verifyNumberOfPublishedEvents(NOTICE_ERROR, 0);
+  }
+
+  @Test
+  void itemLevelRequestExpirationNoticeShouldBeCreatedAndSentButTlrFeatureIsDisabled() {
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java:655]: Isn't this case already covered?

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**